garak/generators/multirest.py

```diff
@@ -0,0 +1,261 @@
+    The generator can handle both JSON and non-JSON responses.
+
+    To set a variable in the response, use the format $n$...$n$, where n is a number.
+
+    For example: 
+        HTTP/2 200 OK
+        Date: Tue, 20 May 2025 05:58:32 GMT
+        Content-Type: application/json
+        Content-Length: 52
+        Server: uvicorn
+        Apigw-Requestid: $2$K2kBRiewPHcEM8A=$2$
+        Access-Control-Allow-Origin: https://platform.dreadnode.io
+
+        {"id":"$1$999999$1$"}]]>
+
+    The above response will set the variable 1 to "999999" and variable 2 to "K2kBRiewPHcEM8A="
+
+    From there you can place these variables in the request body or headers like this:
+        GET /score?id=$1$123$1$ HTTP/2
+        Host: example.com
+        User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:138.0) Gecko/20100101 Firefox/138.0
+        Accept: */*
+        Accept-Language: en-US,en;q=0.5
+        Accept-Encoding: gzip, deflate, br
+        Referer: https://example.com/score
+        Content-Type: application/json
+        Origin: https://example.com
+        Sec-Fetch-Dest: empty 
+        Sec-Fetch-Mode: cors
+        Sec-Fetch-Site: same-site
+        Dnt: 1
+        Sec-Gpc: 1
+        Priority: u=0
+        Te: trailers
+
+    The above request will replace $1$123$1$ with the value of variable 1, which is "999999".
+
+    You can specify where the $INPUT and $OUTPUT variables go in the request and response, as well:
+
+        POST /score HTTP/2
+        Host: example.com
+        User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:138.0) Gecko/20100101 Firefox/138.0
+        Accept: */*
+        Accept-Language: en-US,en;q=0.5
+        Accept-Encoding: gzip, deflate, br
+        Content-Type: application/json
+        Sec-Fetch-Mode: cors
+        Sec-Fetch-Site: same-site
+        Dnt: 1
+        Sec-Gpc: 1
+        Priority: u=0
+        Te: trailers
+
+        {"data":"$INPUT"}
+    
+    Here $INPUT will be replaced with the input value provided to call_model.
+
+        HTTP/2 400 Bad Request
+        Date: Tue, 20 May 2025 05:58:14 GMT
+        Content-Type: application/json
+        Content-Length: 46
+        Server: uvicorn
+        Apigw-Requestid: K2j-dj6KPHcEMvw=
+        Access-Control-Allow-Origin: https://platform.dreadnode.io
+        
+        {"message":"$OUTPUT"}]]>
+
+    Here the location of $OUTPUT in the response body indicates that the output value will be extracted from the response JSON under the "message" key.
+    If the response is not JSON, it will be returned as is.
+    """
+    DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {
+        "reqresp_pairs": [],
+        "variables": {},
+        "burpfile": "./tools/rest/multirest.xml",
+    }
+
+    def __init__(self, name="MultiRestGenerator", config_root=_config):
+        super().__init__(name, config_root=config_root)
```

Comment:
Files provided in the package should in in `garak/data` with pathing matching the package they support.  This allows for hierarchal injection of support file using the `XDG_DATA_HOME/garak/` path for user global replacement.

With such a change this value might be set to `None` to retrieve the standard `data` location file and still allow configuration based override.

```suggestion
        "burpfile": None,
    }

    def __init__(self, name="MultiRestGenerator", config_root=_config):
        super().__init__(name, config_root=config_root)
        if not self.burpfile:
            from garak.data import path as data_path
            
            self.burpfile = data_path / "rest" / "multirest.xml"
```